crates/matrix-sdk-base/src/store/traits.rs

```diff
@@ -1156,7 +1157,16 @@ pub struct QueuedEvent {
     /// If the event couldn't be sent because of an API error, it's marked as
     /// wedged, and won't ever be peeked for sending. The only option is to
     /// remove it.
-    pub is_wedged: bool,
+    pub error: Option<QueueWedgeError>,
+}
+
+impl QueuedEvent {
+    /// If the event couldn't be sent because of an API error, it's marked as
```

Comment:
nit: this is context around what it means to have an error; I'd rather put that near the `error` field itself, and have the doc comment here say something like "returns whether the queued event is wedged (i.e. ran into an unrecoverable error) or not"